app/models/remote_sandbox.rb

```diff
@@ -76,10 +84,21 @@ def send_submission(submission, notify_url)
         submission.save!
       rescue StandardError => e
         puts e
-        raise SandboxUnavailableError
+        begin
+          # Try to parse the response to see if the error originated from within the sandbox.
+          @parsed = JSON.parse e.response
+        rescue StandardError
+          raise SandboxUnavailableError
+        end
+        raise InternalSandboxError.new(@parsed['error'])
       end
     rescue SandboxUnavailableError
       raise
+    rescue InternalSandboxError => e
+      submission.pretest_error = e.object
+      submission.processed = true
```

Comment:
Cannot set processed true here because it will stop the retries.

Remove this and then in the retry  handler make sure that it does not override the pretest error set here.

https://github.com/testmycode/tmc-server/blob/master/lib/submission_processor.rb#L30

The  error message submission processor should be prepended to the pretest_error like
`Tried to process #{Submission.max_attempts_at_processing} times but failed. This is a system error.\n#{previous_pretest_error}`